### PR TITLE
[EDA-1544] execute action 

### PR DESCRIPTION
**Pull request by mbertani-eb**
I renamed the `action_manager` to `execute_action` to clarify what it's doing.

This PR was quite long because `execute_action` is the entry point of the game, and the test for this method is basically an integration test of the game. While I was running the test it started to appears several bugs an error in the transactions methods ( like `make_move` and `modify_score` )

The two test i created were `test_execute_action_move` and `test_execute_action_shoot`
To do  this, I added scenarios, in the `scenarios.py` module but inside a function, because I had problems with the mutability.

I patched  `WumpusGame()` and `Board()` in the tests with `patched_game` and `patched_board` respectly to increase the speed  and performance in testing, because there is a method (`is_valid_hole`) that use a recursive method to create the board.

I add `setup.cfg` file to add flak8 configuration (our linter tool in python) to extend the number of columns allowed to can see better the board's scenarios. 

I also had to re-add the test for `next_turn` and uncomment this method, because in a previous PR had been erased

All the file of `TestShoot` was commented for test that was failing, I fixed all the test and uncomment

### CODE REVIEW

**Review by gojeda-eb — APPROVED** [review]

File: test/test_game.py [hunk 1]
```diff
@@ -280,9 +290,96 @@ def test_penalization_after_invalid_moves(
...
+        ('shoot opponent', 5, 8, 8, WEST, 8, 7, 16100, 4, "     "),
+        ('shoot empty cell', 5, 8, 8, EAST, 8, 9, 1100, 4, "##F##"),
+        ('shoot hole', 5, 8, 8, NORTH, 7, 8, 1100, 4, "  O  "),
+        ('shoot own char', 5, 8, 8, SOUTH, 9, 8, 900, 4, "  B  "),
+    ])
+    def test_execute_action_shoot(self, name,
+                                  initial_arrows,
+                                  from_row, from_col, direction,
+                                  destiny_row, destiny_col,
+                                  expected_score, expected_arrows, expected_destiny_cell):
```
Comment (ceusebi-eb) [inline test/test_game.py:306]: `Destiny` is "the hidden power believed to control future events". You may be referring to `Destination` "the place to which someone or something is going or being sent"
This happens in multiple tests. I'm only leaving a comment here but review all tests

File: test/test_game.py [hunk 2]
```diff
@@ -280,9 +290,96 @@ def test_penalization_after_invalid_moves(
...
+                                 destiny_row, destiny_col,
+                                 expected_score, expected_remaining_characters,
+                                 expected_initial_cell, expected_destiny_cell, ):
+        game = patched_game()
+        board, player_1, player_2 = generate_board_from_move_actio()
+        game.player_1 = player_1
+        game.player_2 = player_2
+        game.current_player = game.player_1
+        game._board._board = board
+        game.current_player.score = 1000
```
Comment (ceusebi-eb) [inline test/test_game.py:343]: This should be a parameter of the test. Also, what happens if the player has 0 points? Are negative scores allowed?

File: test/test_game.py [hunk 3]
```diff
@@ -280,9 +290,96 @@ def test_penalization_after_invalid_moves(
...
+
+        game.execute_action(MOVE, from_row, from_col, direction)
+
+        current_score = game.current_player.score
+        initial_cell = game._board._board[from_row][from_col]
+        destiny_cell = game._board._board[destiny_row][destiny_col]
+        current_remaining_characters = len(game.current_player.characters)
+
+        self.assertEqual(current_score, expected_score)
+        self.assertEqual(initial_cell.to_str(PLAYER_1), expected_initial_cell)
```
Comment (ceusebi-eb) [inline test/test_game.py:353]: Why is this being asserted? Initial cell should be known by the test as it is a param of it.

File: test/test_shoot_method.py [hunk 4]
```diff
@@ -1,125 +1,123 @@
...
+
+    def test_shoot_miss(self):
+        game = patched_game()
+        row = 0
+        col = 1
+        target_cell = game._board._board[row][col]
+        current_player = game.current_player
+        game._board.shoot_miss(row, col, current_player)
+        self.assertEqual(target_cell.arrow, 1)
+        self.assertEqual(current_player.arrows, INITIAL_ARROWS - 1)
```
Comment (ceusebi-eb) [inline test/test_shoot_method.py:87]: It is not clear if the `game._board.shoot_arrow` or `game._board.shoot_miss` action makes you loose an arrow but it's weird that it's being tested twice.
We either test `arrow_decrease` with all possible outcomes of `game._board.shoot_arrow` and don't assert it here or we remove `arrow_decrease` and we test the decrease in these methods that are "aware of the outcome of the shot"

**Review by ceusebi-eb — COMMENTED** [review]
There are a few typos that I would like fixed and some questions around some tests that were uncommented.

Also the PR is too big. My recomendation is:
- First add all the tests that you can in one PR _without refactoring_
- Then make the refactor in a different PR
- Then add all tests that you couldn't add without the refactor (if any) in a third PR.

Examples of things that should have been in a different PR:
- chanages to the flake8 formatting
- refactoring of all tests to use `patched_board` and `patched_game`

File: test/test_game.py (hunk 1 above)
Comment (mbertani-eb) [reply test/test_game.py:306]: Exellent, I'll change it

File: test/test_game.py (hunk 2 above)
Comment (mbertani-eb) [reply test/test_game.py:343]: In the beginning I add `1000` to the parameters, but how it was the same for all the tests I decided to put inside the test. However I'll gonna re-add 
In the game it's allowed to have a negative score, the only way to decrease the score is for invalid moves.

File: test/test_game.py (hunk 3 above)
Comment (ceusebi-eb) [reply test/test_game.py:353]: You can add logs to tests that are only printed when you execute a test in a verbose way. If this is only to debug, then it should be changed for a log

File: test/test_shoot_method.py (hunk 4 above)
Comment (mbertani-eb) [reply test/test_shoot_method.py:87]: I talked with the team, and one of my teammates tell that all the `test_shoot_method.py` module was migrated to the `test_board.py` module, and forgot to erase the file. 
Anyway I checked in  that  module the same thing wasn't happening, and it was all rigth

**Review by ceusebi-eb — APPROVED** [review]

**Pull request merged**